src/python_testing/matter_testing_infrastructure/chip/testing/commissioning.py

```diff
@@ -0,0 +1,137 @@
+#
+#    Copyright (c) 2022-2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+from dataclasses import dataclass
+from typing import List, Optional
+
+from chip import ChipDeviceCtrl  # Needed before chip.FabricAdmin
+from chip import discovery
+from chip.ChipDeviceCtrl import CommissioningParameters
+from chip.clusters import ClusterObjects as ClusterObjects
+from chip.exceptions import ChipStackError
+
+logger = logging.getLogger("matter.python_testing")
+logger.setLevel(logging.INFO)
+
+DiscoveryFilterType = ChipDeviceCtrl.DiscoveryFilterType
+
+
+@dataclass
+class SetupPayloadInfo:
+    filter_type: discovery.FilterType = discovery.FilterType.LONG_DISCRIMINATOR
+    filter_value: int = 0
+    passcode: int = 0
+
+
+@dataclass
+class CommissioningInfo:
+    commissionee_ip_address_just_for_testing: Optional[str] = None
+    commissioning_method: Optional[str] = None
+    thread_operational_dataset: Optional[str] = None
+    wifi_passphrase: Optional[str] = None
+    wifi_ssid: Optional[str] = None
+    # Accepted Terms and Conditions if used
+    tc_version_to_simulate: int = None
```

Comment:
maybe we should fix this if we are touching it: these are obviously `Optional[int]` since they start up as None